opts/secret_test.go

```diff
@@ -8,86 +8,79 @@ import (
-	assert.Check(t, is.Equal("foo", req.File.Name))
-}
-
-func TestSecretOptionsSourceTarget(t *testing.T) {
-	var opt SecretOpt
-
-	testCase := "source=foo,target=testing"
-	assert.NilError(t, opt.Set(testCase))
-
-	reqs := opt.Value()
-	assert.Assert(t, is.Len(reqs, 1))
-	req := reqs[0]
-	assert.Check(t, is.Equal("foo", req.SecretName))
-	assert.Check(t, is.Equal("testing", req.File.Name))
-}
-
-func TestSecretOptionsShorthand(t *testing.T) {
-	var opt SecretOpt
-
-	testCase := "src=foo,target=testing"
-	assert.NilError(t, opt.Set(testCase))
-
-	reqs := opt.Value()
-	assert.Assert(t, is.Len(reqs, 1))
-	req := reqs[0]
-	assert.Check(t, is.Equal("foo", req.SecretName))
-}
-
-func TestSecretOptionsCustomUidGid(t *testing.T) {
-	var opt SecretOpt
-
-	testCase := "source=foo,target=testing,uid=1000,gid=1001"
-	assert.NilError(t, opt.Set(testCase))
-
-	reqs := opt.Value()
-	assert.Assert(t, is.Len(reqs, 1))
-	req := reqs[0]
-	assert.Check(t, is.Equal("foo", req.SecretName))
-	assert.Check(t, is.Equal("testing", req.File.Name))
-	assert.Check(t, is.Equal("1000", req.File.UID))
-	assert.Check(t, is.Equal("1001", req.File.GID))
-}
-
-func TestSecretOptionsCustomMode(t *testing.T) {
-	var opt SecretOpt
-
-	testCase := "source=foo,target=testing,uid=1000,gid=1001,mode=0444"
-	assert.NilError(t, opt.Set(testCase))
-
-	reqs := opt.Value()
-	assert.Assert(t, is.Len(reqs, 1))
-	req := reqs[0]
-	assert.Check(t, is.Equal("foo", req.SecretName))
-	assert.Check(t, is.Equal("testing", req.File.Name))
-	assert.Check(t, is.Equal("1000", req.File.UID))
-	assert.Check(t, is.Equal("1001", req.File.GID))
-	assert.Check(t, is.Equal(os.FileMode(0444), req.File.Mode))
+func TestSecretOptions(t *testing.T) {
+	testCases := []struct {
+		name       string
+		input      string
+		secretName string
+		fileName   string
+		uid        string
+		gid        string
+		fileMode   uint
+	}{
+		{
+			name:       "Simple",
+			input:      "app-secret",
+			secretName: "app-secret",
+			fileName:   "app-secret",
+			uid:        "0",
+			gid:        "0",
+		},
+		{
+			name:       "Source",
+			input:      "source=foo",
+			secretName: "foo",
+			fileName:   "foo",
+		},
+		{
+			name:       "SourceTarget",
+			input:      "source=foo,target=testing",
+			secretName: "foo",
+			fileName:   "testing",
+		},
+		{
+			name:       "Shorthand",
+			input:      "src=foo,target=testing",
+			secretName: "foo",
+			fileName:   "testing",
+		},
+		{
+			name:       "CustomUidGid",
+			input:      "source=foo,target=testing,uid=1000,gid=1001",
+			secretName: "foo",
+			fileName:   "testing",
+			uid:        "1000",
+			gid:        "1001",
+		},
+		{
+			name:       "CustomMode",
+			input:      "source=foo,target=testing,uid=1000,gid=1001,mode=0444",
+			secretName: "foo",
+			fileName:   "testing",
+			uid:        "1000",
+			gid:        "1001",
+			fileMode:   0444,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
same here

```suggestion
		tc := tc
		t.Run(tc.name, func(t *testing.T) {
```

see https://github.com/moby/moby/pull/40115#discussion_r337301763 (and https://blog.golang.org/subtests)